relay-general/src/store/transactions/processor.rs

```diff
@@ -396,6 +414,53 @@ impl Processor for TransactionsProcessor<'_> {
         Ok(())
     }
 
+    /// Validates the length and characters of measurement names.
+    fn process_measurements(
+        &mut self,
+        value: &mut crate::protocol::Measurements,
+        meta: &mut Meta,
+        _state: &ProcessingState<'_>,
+    ) -> ProcessingResult {
+        let measurements = &mut value.0;
+        let mut removed_measurements = Object::new();
+
+        measurements.retain(|key, value| {
+            let measurement = match value.value_mut() {
+                Some(m) => m,
+                None => return false,
+            };
+
+            if !is_valid_metric_name(key) {
+                removed_measurements
+                    .insert(key.clone(), Annotated::new(std::mem::take(measurement)));
+                return false;
+            }
+
+            let name_len = key.len();
+            let unit_len = measurement.unit.value().map(|val| val.to_string().len());
```

Comment:
Measurements without unit get the unit `@none` in metrics MRI, so

```suggestion
            let unit_len = measurement.unit.value().unwrap_or(&MetricUnit::None).to_string().len();
```